### ISSUE TITLE
Parameters not being retrieved

### ISSUE THREAD

**Issue by jgeorge300** [concourse/concourse#1490]
# Bug Report
* Concourse version: 3.3.4
* cf-resource

I have a pipeline configured to retrieve credentials from vault to do a cf deploy. When concourse runs the pipeline, it does not replace the parameters with the values from vault. If I use the parameter elsewhere in the pipeline, the value is retrieved and replaced as expected.

Pipeline snippet: 
```
jobs:
- name: deploy-service-broker
  serial: true
  plan:
  - get: src
  - put: deploy
    params:
      manifest: ((manifest))
      path: src
      environment_variables:
        VAULT_ADDR: ((uri))
```

Error:
```
error invalid manifest path: found 0 files instead of 1 at path: /tmp/build/put/((manifest))
```

**Comment by jtarchie** [comment]
We are going to take a look at `params` on image_resource, too.

**Comment by ayanozturk** [comment]
I have this problem on our Concourse & Vault setup at the moment. Concourse can fetch vault secrets when params are in a resource block but when there is a param under `- put` block it continues without an error and it is using the param definition as string value. On deployed apps, I end up with environment variables like `my_key: ((my_value))` instead of actual secret.

**Comment by ayanozturk** [comment]
Is https://github.com/concourse/atc/commit/08be1e41e7060ecee772656a43567eb60cc3b08e fixed this issue?

**Comment by jtarchie** [comment]
@ayanozturk: Yes. Are you able to validate it works for you?

**Comment by ayanozturk** [comment]
@jtarchie I just upgraded to Concourse 3.6.0 and I can confirm that it can fetch and use params in `- put` block from vault. This issue seems to be fixed in 3.6.0

I also tried to upgrade to 3.7.0 but I had different problems doing that. We are using bosh v1 for deployment and it doesn't like new config keys like `host_key`, `token_signing_key` etc. So I can not confirm if it is still working on 3.7 release.
